stan/math/rev/fun/variance.hpp

```diff
@@ -67,6 +67,32 @@ var variance(const EigMat& m) {
   return {internal::calc_variance(mat.size(), mat.data())};
 }
 
+/**
+ * Return the sample variance of the var_value matrix
+ * Raise domain error if size is not greater than zero.
+ *
+ * @param[in] x a input
+ * @return sample variance of input
+ */
+template <typename T, require_var_matrix_t<T>* = nullptr>
+inline var variance(const T& x) {
+  check_nonzero_size("variance", "x", x);
+
+  if (x.size() == 1) {
+    return 0.0;
+  }
```

Comment:
[optional] Should we put these in `unlikely()` calls? idt they matter that much so no biggie

Reply:
I'll leave it to the branch predictor.